### PR TITLE
add registration form

### PR DESCRIPTION
**Pull request by Olesya987**

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,38 +1,100 @@
 .App {
   text-align: center;
+  
+}
+body{
```
Comment (usatovaDasha) [inline src/App.css]: spaces

File: src/App.css [hunk 2]
```diff
@@ -1,38 +1,100 @@
...
+  background-color: #d0eaff;
+}
```
Comment (usatovaDasha) [inline src/App.css]: add empty line

File: src/App.css [hunk 3]
```diff
@@ -1,38 +1,100 @@
...
+h2{
+  margin: 20px;
 }
 
 .App-logo {
```
Comment (usatovaDasha) [inline src/App.css]: you need app-logo?

File: src/App.css [hunk 4]
```diff
@@ -1,38 +1,100 @@
...
+  color: rgb(0, 0, 0);
+  box-shadow: 0px 7px 19px 0px #4b5e75;
 }
 
 .App-link {
   color: #61dafb;
 }
 
-@keyframes App-logo-spin {
+.main-reg{
```
Comment (usatovaDasha) [inline src/App.css]: spaces

File: src/App.css [hunk 5]
```diff
@@ -1,38 +1,100 @@
...
+  width: 100%;
+  height: 85vh;
+}
+
+form{
+  display: flex;
+  flex-direction: column;
+  align-items: flex-end;
+  width: 100%;
+}
```
Comment (usatovaDasha) [inline src/App.css]: add empty lines

File: src/App.css [hunk 6]
```diff
@@ -1,38 +1,100 @@
...
+  box-shadow: 7px 7px 19px 0px #4b5e75;
+}
+.input-form-reg{
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+}
+
+.MuiButton-outlinedPrimary {
+  margin: 8px !important;
```
Comment (usatovaDasha) [inline src/App.css]: don't use important, using your className with class material

File: src/App.js [hunk 7]
```diff
@@ -1,23 +1,58 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from "react";
```
Comment (usatovaDasha) [inline src/App.js:1]: imports style, 
i send in general channel link on presentation

File: src/App.js [hunk 8]
```diff
@@ -1,23 +1,58 @@
...
+import medical from "./medical-2.png";
+import { Route, Switch, useHistory, Redirect } from "react-router-dom";
+import Button from "@material-ui/core/Button";
 
 function App() {
+  const history = useHistory();
+  const [authReg, setAuthReg] = useState({
+    text: "Вход в систему",
+    flag: false,
+  });
```
Comment (usatovaDasha) [inline src/App.js:18]: add empty line

File: src/App.js [hunk 9]
```diff
@@ -1,23 +1,58 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
+        <img className="App-logo" src={icon}></img>
+        <h1>{authReg.text}</h1>
+        {authReg.flag ? (
+          <div>
+            {/* <h4>{history.pop()}</h4> */}
```
Comment (usatovaDasha) [inline src/App.js]: remove comments

File: src/App.js [hunk 10]
```diff
@@ -1,23 +1,58 @@
...
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
+        <img className="App-logo" src={icon}></img>
+        <h1>{authReg.text}</h1>
+        {authReg.flag ? (
```
Comment (usatovaDasha) [inline src/App.js]: {authReg.flag && <div> LogOut </>}

File: src/App.js [hunk 11]
```diff
@@ -1,23 +1,58 @@
...
+        )} */}
+
+        <Switch>
+          <Route path="/authorization">
+            <Auth setAuthReg={setAuthReg} />
+          </Route>
+          <Route path="/registration">
+            <Reg setAuthReg={setAuthReg} />
+          </Route>
+          <Route path="/appointments/:login">
```
Comment (usatovaDasha) [inline src/App.js]: why you need dynamic params?

File: src/Auth.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
+import React, { useState } from "react";
+import { Route, Switch, useHistory, Redirect } from "react-router-dom";
```
Comment (usatovaDasha) [inline src/Auth.js]: please add empty lines

File: src/Auth.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+const Auth = ({ setAuthReg }) => {
+  const history = useHistory();
+  return (
+    <div>
+      <h2>Войти в систему</h2>
+      <form>
+        <div>
+          <label>Login:</label>
+          <input type="text" id="input-login" name="input-login" />
```
Comment (usatovaDasha) [inline src/Auth.js]: you using material-ui?

File: src/Auth.js [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+          <label>Password:</label>
+          <input type="text" id="input-password" name="input-password" />
+        </div>
+        <button>Войти</button>
+      </form>
+      <a
```
Comment (usatovaDasha) [inline src/Auth.js]: don't using a in React

File: src/Auth.js [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+        onClick={() => {
+          setAuthReg({
```
Comment (usatovaDasha) [inline src/Auth.js]: in function

File: src/Reg.js [hunk 16]
```diff
@@ -0,0 +1,231 @@
...
+import clsx from "clsx";
+import {
+  alpha,
+  ThemeProvider,
+  withStyles,
+  makeStyles,
+  createTheme,
+} from "@material-ui/core/styles";
+import IconButton from "@material-ui/core/IconButton";
+import Input from "@material-ui/core/Input";
```
Comment (usatovaDasha) [inline src/Reg.js]: core in one import

File: src/Reg.js [hunk 17]
```diff
@@ -0,0 +1,231 @@
...
+import Visibility from "@material-ui/icons/Visibility";
+import VisibilityOff from "@material-ui/icons/VisibilityOff";
+import Button from "@material-ui/core/Button";
+
+const Reg = ({ setAuthReg }) => {
+  const history = useHistory();
+
+  const [user, setUser] = useState({});
+  const [passCorrect, setPass] = useState(false);
+  const useStyles = makeStyles((theme) => ({
```
Comment (usatovaDasha) [inline src/Reg.js]: using scss file

File: src/Reg.js [hunk 18]
```diff
@@ -0,0 +1,231 @@
...
+    },
+    textField: {
+      width: "auto",
+    },
+  }));
+
+  const classes = useStyles();
+  const [values, setValues] = React.useState({
+    login: "",
+    amount: "",
```
Comment (usatovaDasha) [inline src/Reg.js]: amount?

File: src/Reg.js [hunk 19]
```diff
@@ -0,0 +1,231 @@
...
+    password: "",
+    weight: "",
```
Comment (usatovaDasha) [inline src/Reg.js]: weight?

File: src/Reg.js [hunk 20]
```diff
@@ -0,0 +1,231 @@
...
+    setValues({ ...values, [prop]: event.target.value });
+  };
+
+  const handleClickShowPassword = () => {
+    setValues({ ...values, showPassword: !values.showPassword });
+  };
+
+  const handleClickShowPasswordRepeat = () => {
+    setValues({ ...values, showPasswordRepeat: !values.showPasswordRepeat });
+  };
```
Comment (usatovaDasha) [inline src/Reg.js]: add empty line

File: src/Reg.js [hunk 21]
```diff
@@ -0,0 +1,231 @@
...
+  const checkPass = () => {
+    let flag1 = true;
+    let flag3 = false;
+    const regexp = /((?=.*[0-9])(?=.*[a-zA-Z]).{6,})/g;
+    if (regexp.test(values.password)) {
+      flag3 = true;
+    }
+    if (/[а-яА-Я]/.test(values.password)) flag1 = false;
```
Comment (usatovaDasha) [inline src/Reg.js]: russian lang?

File: src/Reg.js [hunk 22]
```diff
@@ -0,0 +1,231 @@
...
+
+    return flag1 && flag3;
```
Comment (usatovaDasha) [inline src/Reg.js]: return regexp.test(values.password) && (/[а-яА-Я]/.test(values.password));

File: src/Reg.js [hunk 23]
```diff
@@ -0,0 +1,231 @@
...
+  };
+
+  const subUser = async (e) => {
+    e.preventDefault();
+    console.log(checkPass());
```
Comment (usatovaDasha) [inline src/Reg.js]: remove

File: src/Reg.js [hunk 24]
```diff
@@ -0,0 +1,231 @@
...
+    const formData = new FormData(e.target);
+    if (
+      formData.get("input-login").length >= 6 &&
```
Comment (usatovaDasha) [inline src/Reg.js]: only 6 symbols

File: src/Reg.js [hunk 25]
```diff
@@ -0,0 +1,231 @@
...
+      checkPass() &&
+      values.password.length >= 6 &&
+      values.password === values.passwordRepeat
+    ) {
+      await axios
+        .post("http://localhost:8080/user/post", {
+          login: formData.get("input-login"),
```
Comment (usatovaDasha) [inline src/Reg.js]: you check values.password but using formData.get("input-login")

File: src/Reg.js [hunk 26]
```diff
@@ -0,0 +1,231 @@
...
+          password: formData.get("input-password"),
+        })
+        .then((res) => {
+          history.push(`/appointments/${res.data.login}`)
```
Comment (usatovaDasha) [inline src/Reg.js]: ;

File: src/Reg.js [hunk 27]
```diff
@@ -0,0 +1,231 @@
...
+        amount: "",
+        password: "",
+        weight: "",
+        weightRange: "",
+        showPassword: false,
+        showPasswordRepeat: false,
+        passwordRepeat: "",
+      });
+    } else {
+      alert(
```
Comment (usatovaDasha) [inline src/Reg.js]: don't use alert, using snackbar

File: src/Reg.js [hunk 28]
```diff
@@ -0,0 +1,231 @@
...
+      );
+    }
+  };
+
+  return (
+    <div className="main-reg-fild">
+      <h2>Регистрация</h2>
+      <form onSubmit={subUser}>
+        <div className="input-form-reg">
+          {/* <label>Login:</label>
```
Comment (usatovaDasha) [inline src/Reg.js]: remove comments

File: src/Reg.js [hunk 29]
```diff
@@ -0,0 +1,231 @@
...
+              type="text"
+              variant="outlined"
+              style={{ width: "100%" }}
+              value={values.login}
+              onChange={handleChange("login")}
+            />
+          </div>
+
+          <FormControl
+            className={clsx(classes.margin, classes.textField)}
```
Comment (usatovaDasha) [inline src/Reg.js]: scss file, and your className

File: src/Reg.js [hunk 30]
```diff
@@ -0,0 +1,231 @@
...
+          </FormControl>
+        </div>
+        <Button variant="outlined" color="primary" type="none">
+          Зарегистрироваться
+        </Button>
+        {/* <button>Зарегистрироваться</button> */}
+        <div
+          className="reg-text"
+          onClick={() => {
+            setAuthReg({
```
Comment (usatovaDasha) [inline src/Reg.js]: in function

File: src/appointments.js [hunk 31]
```diff
@@ -0,0 +1,32 @@
+import React, { useState } from "react";
```
Comment (usatovaDasha) [inline src/appointments.js]: empty lines

File: src/App.js [hunk 32]
```diff
@@ -1,23 +1,58 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from "react";
+import logo from "./logo.svg";
+import cross from "./cross.png";
```
Comment (usatovaDasha) [inline src/App.js]: images in folder

File: src/App.css (hunk 3 above)
Comment (Olesya987) [reply src/App.css]: yes, i used it to hospital logo

File: src/App.js (hunk 7 above)
Comment (usatovaDasha) [reply src/App.js:1]: !!!!!

File: src/Reg.js [hunk 33]
```diff
@@ -1,14 +1,7 @@
...
 import axios from "axios";
-import clsx from "clsx";
-import {
-  alpha,
-  ThemeProvider,
-  withStyles,
-  makeStyles,
-  createTheme,
-} from "@material-ui/core/styles";
+import "./Reg.scss";
```
Comment (usatovaDasha) [inline src/Reg.js]: imports style

File: src/Reg.js [hunk 34]
```diff
@@ -92,24 +75,31 @@ const Reg = ({ setAuthReg }) => {
...
           setAuthReg({
             text: "Приемы",
-            flag:true,
+            flag: true,
+          });
+          setValues({
+            login: "",
+            amount: "",
+            password: "",
+            weight: "",
```
Comment (usatovaDasha) [inline src/Reg.js]: please correct names

File: src/Reg.js [hunk 35]
```diff
@@ -92,24 +75,31 @@ const Reg = ({ setAuthReg }) => {
...
+            weightRange: "",
+            showPassword: false,
+            showPasswordRepeat: false,
+            passwordRepeat: "",
           });
+        })
+        .catch((err) => {
+          if (err.response.status === 421) {
+            alert("Пользователь с таким логином уже зарегистрирован");
```
Comment (usatovaDasha) [inline src/Reg.js]: using snackbar

File: src/Reg.js (hunk 27 above)
Comment (usatovaDasha) [reply src/Reg.js]: !!!!

File: src/Reg.scss [hunk 36]
```diff
@@ -0,0 +1,70 @@
+.root {
+  display: flex;
+  flex-wrap: wrap;
+}
+
+.margin {
+  margin: 8px !important;
```
Comment (usatovaDasha) [inline src/Reg.scss]: don't use important

File: src/Reg.scss [hunk 37]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+.main-reg-fild {
+  width: 30%;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  background-color: #ffffff;
+  height: fit-content;
+  padding: 30px 30px;
```
Comment (usatovaDasha) [inline src/Reg.scss]: padding: 30px;

File: src/App.js (hunk 10 above)
Comment (usatovaDasha) [reply src/App.js]: !!!!!

File: src/Reg/Reg.js [hunk 38]
```diff
@@ -1,30 +1,39 @@
...
-import OutlinedInput from "@material-ui/core/OutlinedInput";
-import InputLabel from "@material-ui/core/InputLabel";
-import InputAdornment from "@material-ui/core/InputAdornment";
-import FormControl from "@material-ui/core/FormControl";
-import TextField from "@material-ui/core/TextField";
-import Visibility from "@material-ui/icons/Visibility";
-import VisibilityOff from "@material-ui/icons/VisibilityOff";
-import Button from "@material-ui/core/Button";
+
+function Alert(props) {
```
Comment (usatovaDasha) [inline src/Reg/Reg.js]: es6

File: src/Reg/Reg.js [hunk 39]
```diff
@@ -42,20 +51,27 @@ const Reg = ({ setAuthReg }) => {
     setValues({ ...values, showPasswordRepeat: !values.showPasswordRepeat });
   };
   const checkPass = () => {
-    let flag1 = true;
-    let flag3 = false;
     const regexp = /((?=.*[0-9])(?=.*[a-zA-Z]).{6,})/g;
-    if (regexp.test(values.password)) flag3 = true;
-    if (/[а-яА-Я]/.test(values.password)) flag1 = false;
+    return regexp.test(values.password) && /[а-яА-Я]/.test(values.password);
```
Comment (usatovaDasha) [inline src/Reg/Reg.js]: russian language?

File: src/Reg/Reg.js [hunk 40]
```diff
@@ -75,35 +91,37 @@ const Reg = ({ setAuthReg }) => {
           password: formData.get("input-password"),
         })
         .then((res) => {
-          setUser({
-            login: res.data.login,
-            token: res.data.token,
-          });
-          history.push(`/appointments/${res.data.login}`);
+          setUser(res.data.user);
+          history.push(`/appointments/${user.login}`);
```
Comment (usatovaDasha) [inline src/Reg/Reg.js]: user.login need?

File: src/Reg/Reg.js [hunk 41]
```diff
@@ -91,11 +87,12 @@ const Reg = ({ setAuthReg }) => {
           password: formData.get("input-password"),
         })
         .then((res) => {
-          setUser(res.data.user);
-          history.push(`/appointments/${user.login}`);
+          history.push(`/appointments`);
           setAuthReg({
             text: "Приемы",
             flag: true,
+            login: res.data.login,
```
Comment (usatovaDasha) [inline src/Reg/Reg.js]: destructurization

File: src/Reg/Reg.js [hunk 42]
```diff
@@ -12,13 +12,13 @@ import {
   Snackbar,
 } from "@material-ui/core";
 import { Visibility, VisibilityOff } from "@material-ui/icons";
-import MuiAlert from '@material-ui/lab/Alert';
+import MuiAlert from "@material-ui/lab/Alert";
 import CloseIcon from "@material-ui/icons/Close";
 import "./Reg.scss";
 
-function Alert(props) {
+const Alert = (props) => {
```
Comment (usatovaDasha) [inline src/Reg/Reg.js]: message

File: src/Reg/Reg.js [hunk 43]
```diff
@@ -207,7 +200,7 @@ const Reg = ({ setAuthReg }) => {
         open={state.open}
         autoHideDuration={13000}
         onClose={handleClose}
-        message={state.text}
+        // message={state.text}
```
Comment (usatovaDasha) [inline src/Reg/Reg.js]: remove

File: src/Reg/Reg.js [hunk 44]
```diff
@@ -87,12 +83,13 @@ const Reg = ({ setAuthReg }) => {
           password: formData.get("input-password"),
         })
         .then((res) => {
+          const { login, token } = res.data;
           history.push(`/appointments`);
           setAuthReg({
             text: "Приемы",
```
Comment (usatovaDasha) [inline src/Reg/Reg.js:86]: {
            text: "Приемы",
            flag: true,
            login,
            token
          }

File: src/Auth/Auth.js [hunk 45]
```diff
@@ -3,6 +3,9 @@ import { Route, Switch, useHistory, Redirect } from "react-router-dom";
 
 const Auth = ({ setAuthReg }) => {
   const history = useHistory();
+
```
Comment (usatovaDasha) [inline src/Auth/Auth.js:20]: remove empty lines

**Pull request merged**